scikit_tt/data_driven/regression.py

```diff
@@ -296,7 +297,7 @@ def __arr_construct_stack_left(i, stack_left, x_data, basis_list, solution):
         m = x_data.shape[1]
 
         # contract previous stack element with solution and TDT cores
-        stack_left[i] = np.array([basis_list[i-1][k](x_data) for k in range(n)])
+        stack_left[i] = np.array([[basis_list[i-1][k](x_data[:, j]) for j in range(m)] for k in range(n)])
```

Comment:
Unfortunately you cannot feed the new classes of basis functions data arrays. Only single points.